File: scrapegraphaiapisdk/feedback.py
"""
This module provides functionality to send feedback to the ScrapeGraph AI API.

It includes a function to send feedback messages along with the necessary API key
and handles responses and errors appropriately.
"""

import requests
import json

def feedback(api_key: str, feedback: str) -> str:
    """Send feedback to the API.

    Args:
        api_key (str): Your ScrapeGraph AI API key.
        feedback (str): The feedback message to send.

    Returns:
        str: Response from the API in JSON format.
    """
    endpoint = "https://sgai-api.onrender.com/api/v1/feedback"
    headers = {
        "accept": "application/json",
        "SGAI-API-KEY": api_key,
        "Content-Type": "application/json"
    }
    
    feedback_data = {"feedback": feedback}  # Prepare the feedback data

    try:
        response = requests.post(endpoint, headers=headers, json=feedback_data)
        response.raise_for_status()
    except requests.exceptions.HTTPError as http_err:
        return json.dumps({"error": "HTTP error occurred", "message": str(http_err), "status_code": response.status_code})
    except requests.exceptions.RequestException as e:
        return json.dumps({"error": "An error occurred", "message": str(e)})
    
    return response.text

File: scrapegraph-js/feedback.js
import axios from 'axios';

/**
 * Send feedback to the API.
 * 
 * @param {string} apiKey - Your ScrapeGraph AI API key
 * @param {string} requestId - The request ID associated with the feedback
 * @param {number} rating - The rating score
 * @param {string} feedbackText - The feedback message to send
 * @returns {Promise<string>} Response from the API in JSON format
 */
export async function feedback(apiKey, requestId, rating, feedbackText) {
  const endpoint = "https://sgai-api.onrender.com/api/v1/feedback";
  const headers = {
    "accept": "application/json",
    "SGAI-API-KEY": apiKey,
    "Content-Type": "application/json"
  };

  const feedbackData = {
    request_id: requestId,
    rating: rating,
    feedback_text: feedbackText
  };

  try {
    const response = await axios.post(endpoint, feedbackData, { headers });
    return JSON.stringify(response.data);
  } catch (error) {
    if (error.response) {
      return JSON.stringify({
        error: "HTTP error occurred",
        message: error.message,
        status_code: error.response.status
      });
    }
    return JSON.stringify({
      error: "An error occurred",
      message: error.message
    });
  }
} 